File: knowledge-vault-ui/src/app/models/reg/userdetails.ts
export class UserDetails {
    firstname: string;
    lastname: string;
    username: string;
    role: string;
    password: string;

    constructor(fn, ln, un, r, pwd) {
        this.firstname = fn;
        this.lastname = ln;
        this.username = un;
        this.role = r;
        this.password = pwd;
    }
}

File: user-registration-service/src/main/java/com/stackroute/domain/UserDetails.java
package com.stackroute.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Transient;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Arrays;
@AllArgsConstructor
@NoArgsConstructor
@Data
@Document
public class UserDetails {

    private String firstname;
    private String lastname;
    @Id
    private String username;

    private String role;
    @Transient
    private String password;

}
